apollo-router/tests/common.rs

```diff
@@ -221,36 +278,38 @@ impl IntegrationTest {
             .expect("must be able to write config");
     }
 
-    pub async fn run_query(&self) -> (String, reqwest::Response) {
+    pub fn run_query(&self) -> impl std::future::Future<Output = (String, reqwest::Response)> {
```

Comment:
NIT: why convert from an async function to a sync function returning a future? This is functionally the same, no issue there, just wondering why

Reply:
It allows the function to be called without moving `self` into the async block. The call to assert `self.router.is_some()` happens before the async bock.

Reply:
Without this the borrow checker complains.